Project/Assets/ML-Agents/Examples/Match3/Scripts/Match3Drawer.cs

```diff
@@ -9,37 +10,108 @@ public class Match3Drawer : MonoBehaviour
         {
-          Color.red,
-          Color.green,
-          Color.blue,
-          Color.cyan,
-          Color.magenta,
-          Color.yellow,
-          Color.gray,
-          Color.black,
+            Color.red,
+            Color.green,
+            Color.blue,
+            Color.cyan,
+            Color.magenta,
+            Color.yellow,
+            Color.gray,
+            Color.black,
         };
 
         private static Color s_EmptyColor = new Color(0.5f, 0.5f, 0.5f, .25f);
 
+        public Dictionary<(int, int), Match3TileSelector> tilesDict = new Dictionary<(int, int), Match3TileSelector>();
+        public float CubeSpacing = 1.25f;
+        public GameObject TilePrefab;
+
+        private bool m_Initialized;
+        private Match3Board m_Board;
+
+        void Awake()
+        {
+            if (!m_Initialized)
+            {
+                InitializeDict();
+            }
+        }
+
+        void InitializeDict()
+        {
+            m_Board = GetComponent<Match3Board>();
+            foreach (var item in tilesDict)
+            {
+                if (item.Value)
+                {
+                    DestroyImmediate(item.Value.gameObject);
+                }
+            }
+
+            tilesDict.Clear();
+
+            for (var i = 0; i < m_Board.Rows; i++)
+            {
+                for (var j = 0; j < m_Board.Columns; j++)
+                {
+                    var go = Instantiate(TilePrefab, transform.position, Quaternion.identity, transform);
+                    go.name = $"r{i}_c{j}";
+                    tilesDict.Add((i, j), go.GetComponent<Match3TileSelector>());
+                }
+            }
+
+            m_Initialized = true;
+        }
+
+        void Update()
+        {
+            if (!m_Board)
+            {
+                m_Board = GetComponent<Match3Board>();
+            }
+
+            if (!m_Initialized)
+            {
+                InitializeDict();
+            }
+
+            for (var i = 0; i < m_Board.Rows; i++)
+            {
+                for (var j = 0; j < m_Board.Columns; j++)
+                {
+                    var value = m_Board.Cells != null ? m_Board.GetCellType(i, j) : Match3Board.k_EmptyCell;
+                    var pos = new Vector3(j, i, 0);
+                    pos *= CubeSpacing;
+
+                    var specialType = m_Board.Cells != null ? m_Board.GetSpecialType(i, j) : 0;
+                    tilesDict[(i, j)].transform.position = transform.TransformPoint(pos);
+                    tilesDict[(i, j)].SetActiveTile(specialType, value);
+                }
+            }
+        }
 
         void OnDrawGizmos()
         {
             // TODO replace Gizmos for drawing the game state with proper GameObjects and animations.
             var cubeSize = .5f;
-            var cubeSpacing = .75f;
-            var matchedWireframeSize = .5f * (cubeSize + cubeSpacing);
+            var matchedWireframeSize = .5f * (cubeSize + CubeSpacing);
```

Comment:
Do we even need the gizmos code anymore?

Reply:
Even if we want to keep it around, you can remove the TODO above :)

Reply:
changed: removed the todo. also, the gizmos still show some of the debug features you made. there's still value there so i'd say we can leave for now